runtime/services/catalog/migrations.go

```diff
@@ -452,7 +459,10 @@ func (s *Service) collectMigrationItems(
 	// TODO: is there a better way to do this?
 	tempDag := dag.NewDAG()
 	for name, migration := range migrationMap {
-		tempDag.Add(name, migration.NormalizedDependencies)
+		_, err := tempDag.Add(name, migration.NormalizedDependencies)
+		if err != nil {
+			return nil, err
```

Comment:
We should not be returning error for the entire reconcile. But just for the artifact that caused this. There is a check later where if strict=true returns error.

This applies to all places where we handle the error from Add